drivers/network/netio/netio.c

```diff
@@ -0,0 +1,896 @@
+                            if (InputBuffer == NULL)
+                            {
+                                DbgPrint("WskControlSocket: Need an InputBuffer for this operation\n");
+                                status = STATUS_INVALID_PARAMETER;
+                                break;
+                            }
+                            if (InputSize < 4)
+                            {
+                                DbgPrint("WskControlSocket: InputBuffer too small for this operation\n");
+                                status = STATUS_INVALID_PARAMETER;
+                                break;
+                            }
+                            int flag = *(int *)InputBuffer;
+                            if (flag != 0)
+                            {
+                                s->Flags |= ControlCode;
+                            }
+                            else
+                            {
+                                s->Flags &= ~ControlCode;
+                            }
+                            status = STATUS_SUCCESS;
+
+                            break;
+                            /* Windows specific. This sets the mask for callback functions: */
+                        case SO_WSK_EVENT_CALLBACK:
+                            if (InputBuffer == NULL)
+                            {
+                                DbgPrint("WskControlSocket: Need an InputBuffer for this operation\n");
+                                status = STATUS_INVALID_PARAMETER;
+                                break;
+                            }
+                            if (InputSize < sizeof(WSK_EVENT_CALLBACK_CONTROL))
+                            {
+                                DbgPrint("WskControlSocket: InputBuffer too small for this operation\n");
+                                status = STATUS_INVALID_PARAMETER;
+                                break;
+                            }
+                            WSK_EVENT_CALLBACK_CONTROL *c = (WSK_EVENT_CALLBACK_CONTROL *) InputBuffer;
+
+                            s->CallbackMask = c->EventMask;
+                            /* TODO: and start listening here? */
+
+                            status = STATUS_SUCCESS;
+                            break;
+
+                        default:
+                            DbgPrint("WskControlSocket: ControlCode %d Not implemented\n", ControlCode);
+                    }
+                    break;
+                default:
+                    DbgPrint("WskControlSocket: Level %d Not implemented\n", Level);
+            }
+            break;
+
+        case WskGetOption:
+        case WskIoctl:
+        default:
+            DbgPrint("WskControlSocket: Option %d Not implemented\n", RequestType);
+    }
+    if (Irp != NULL)
+    {
+        Irp->IoStatus.Status = status;
+/* ??		Irp->IoStatus.Information = Irp->IoStatus.Information; */
```

Comment:
When do you want to add this? Why isn't it added yet? May deserve a comment.

Reply:
Ok I removed the line.